app/src/main/java/edu/wpi/first/shuffleboard/app/prefs/AppPreferences.java

```diff
@@ -34,7 +41,21 @@ public static AppPreferences getInstance() {
   public ImmutableList<Property<?>> getProperties() {
     return ImmutableList.of(
         theme,
-        defaultTileSize
+        defaultTileSize,
+        server,
+        port
+    );
+  }
+
+  /**
+   * Gets a read-only list of all the preference properties.
+   */
+  public ImmutableList<Property<?>> getFlushableProperties() {
```

Comment:
this is a bad interface—it requires the caller to know whether it wants flushable properties or not. In reality, we want to mix and match the properties. For example, theme should change in real time, not just when you close the window.